batch_seqr_loader/batch_workflow.py

```diff
@@ -9,102 +9,63 @@
 import json
 import logging
 import os
+import re
 import shutil
 import subprocess
+import sys
 import tempfile
-import hashlib
-from collections import defaultdict
 from os.path import join, dirname, abspath, splitext
-from typing import Optional, List, Tuple, Set, Iterable
+from typing import Optional, List, Tuple, Set, Dict, Union
 import pandas as pd
 import click
 import hailtop.batch as hb
 from analysis_runner import dataproc
 from hailtop.batch.job import Job
-from find_inputs import find_inputs
-from utils import file_exists
-
-GATK_VERSION = '4.2.0.0'
-GATK_CONTAINER = (
-    f'australia-southeast1-docker.pkg.dev/cpg-common/images/gatk:{GATK_VERSION}'
+from models.models.analysis import Analysis
+from sample_metadata import (
+    SampleApi,
+    AnalysisApi,
+    AnalysisModel,
+    Sample,
+    AnalysisUpdateModel,
 )
-PICARD_CONTAINER = f'us.gcr.io/broad-gotc-prod/picard-cloud:2.23.8'
-BAZAM_CONTAINER = f'australia-southeast1-docker.pkg.dev/cpg-common/images/bazam:v2'
-SOMALIER_CONTAINER = 'brentp/somalier:latest'
-PEDDY_CONTAINER = 'quay.io/biocontainers/peddy:0.4.8--pyh5e36f6f_0'
-
-NUMBER_OF_HAPLOTYPE_CALLER_INTERVALS = 50
-NUMBER_OF_DATAPROC_WORKERS = 50
-NUMBER_OF_GENOMICS_DB_INTERVALS = 10
-
-REF_BUCKET = 'gs://cpg-reference/hg38/v1'
-REF_FASTA = join(REF_BUCKET, 'Homo_sapiens_assembly38.fasta')
-DBSNP_VCF = join(REF_BUCKET, 'Homo_sapiens_assembly38.dbsnp138.vcf')
-UNPADDED_INTERVALS = join(REF_BUCKET, 'hg38.even.handcurated.20k.intervals')
-SOMALIER_SITES = join(REF_BUCKET, 'sites.hg38.vcf.gz')
-
-DATAPROC_PACKAGES = [
-    'seqr-loader',
-    'click',
-    'google',
-    'slackclient',
-    'fsspec',
-    'sklearn',
-    'gcloud',
-]
+
+from vqsr import make_vqsr_jobs
+import utils
+
 
 logger = logging.getLogger(__file__)
 logging.basicConfig(format='%(levelname)s (%(name)s %(lineno)s): %(message)s')
 logger.setLevel(logging.INFO)
 
 
+sapi = SampleApi()
+aapi = AnalysisApi()
+
+
 @click.command()
 @click.option(
-    '--gvcf',
-    'gvcfs',
-    multiple=True,
-    help='Glob path to input GVCF files.C orresponding tbi index files are expected '
-    'exist',
-)
-@click.option(
-    '--bam',
-    '--cram',
-    'crams',
-    multiple=True,
-    help='Glob path to input BAM or CRAM files. Corresponding bai/crai indices are '
-    'expected to existVariants will be called with GATK HaplotypeCaller',
-)
-@click.option(
-    '--bam-to-realign',
-    '--cram-to-realign',
-    '--fastq-to-realign',
-    '--data-to-realign',
-    'data_to_realign',
-    multiple=True,
-    help='Glob path with input BAM/CRAM/FASTQ files that need re-alignment. '
-    'For BAMs and CRAMs, corresponding bai/crai indices are expected to exist. '
-    'Files will be (re-)aligned with BWA, and variants will be called with '
-    'GATK HaplotypeCaller',
+    '--sm-db-name',
+    'sm_db_name',
+    default='seqr',
```

Comment:
We could remove this default given we're creating SM projects for each subproject.

Reply:
Leaving this option as `--test-sm-db-name` just to support testing with `vladdev`. The default value is `acute-care` for now, but will rename after the DB reorganisation :)